Automattic/jetpack
# eslint-config-target-es: Support flat config (eslint v9)

## Proposed changes:

This adds new entry points for the flat configs corresponding to the existing eslintrc entry points, and for good measure adds explicit entry points for the eslintrc versions as well. When we next do a major version bump, we might swap the "default" entry points from eslintrc to flat config.

### Other information:


## Jetpack product discussion


None

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* CI happy? That's a pretty good test eslintrc still works.
* Proofread readme changes.
* To test the flat config, in a fresh directory somewhere you should be able to `pnpm add file:/path/to/jetpack-monorepo/projects/js-packages/eslint-config-target-es` (which should also bring in eslint 9.4.0 as an automatic peer dep), set up a flat config (see the readme), and run eslint and have it still work.
